packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation.js

```diff
@@ -0,0 +1,85 @@
+import { useCallback } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+import { useFieldState } from 'informed';
+
+import { useAppContext } from '../../../context/app';
+import { useCartContext } from '../../../context/cart';
+
+/**
+ *
+ * @param {Function} props.onSave callback to be called when user clicks review order button
+ * @param {Function} props.resetReviewOrderButtonClicked callback to reset the review order button flag
+ * @param {DocumentNode} props.queries.getCheckoutStepQuery query to get the current checkout page step
+ *
+ * @returns {
+ *   doneEditing: Boolean,
+ *   isEditModalHidden: Boolean,
+ *   showEditModal: Function,
+ *   hideEditModal: Function,
+ *   handlePaymentError: Function,
+ *   handlePaymentSuccess: Function,
+ *   currentSelectedPaymentMethod: String,
+ *   checkoutStep: Number,
+ *
+ * }
+ */
+export const usePaymentInformation = props => {
+    const { queries, onSave, resetReviewOrderButtonClicked } = props;
+    const { getCheckoutStepQuery } = queries;
+
+    /**
+     * Definitions
+     */
+
+    const [{ drawer }, { toggleDrawer, closeDrawer }] = useAppContext();
+    const isEditModalActive = drawer === 'edit.payment';
+    const [{ cartId }] = useCartContext();
+    const { value: currentSelectedPaymentMethod } = useFieldState(
+        'selectedPaymentMethod'
+    );
+
+    /**
+     * Query Fetches
+     */
+
+    const { data: checkoutStepData } = useQuery(getCheckoutStepQuery, {
+        variables: cartId
+    });
+
+    const checkoutStep = checkoutStepData
+        ? checkoutStepData.cart.checkoutStep
+        : 1;
+
+    /**
+     * Helper Functions
+     */
+
+    const showEditModal = useCallback(() => {
+        toggleDrawer('edit.payment');
+    }, [toggleDrawer]);
+
+    const hideEditModal = useCallback(() => {
+        closeDrawer('edit.payment');
+    }, [closeDrawer]);
+
+    const handlePaymentSuccess = useCallback(() => {
+        if (onSave) {
+            onSave();
+        }
+    }, [onSave]);
+
+    const handlePaymentError = useCallback(() => {
+        resetReviewOrderButtonClicked();
+    }, [resetReviewOrderButtonClicked]);
+
+    return {
+        doneEditing: checkoutStep > 3,
```

Comment:
I remember mentioning this in the past, but the presence of data is not the same as the completion of a step in case of payment information. Even if we have all the data we need, payment information can actually result in an incomplete step due to multiple submit steps internally. That is the reason why I am using `checkoutStep` to determine `doneEditing` and that is also the reason why I do not call it `hasData` because they are diff. Like you mentioned I can use a local variable to determine this but I was asked not to create local state when you can derive it. Not sure which one to follow. Honestly, I don't see a reason why it is bad to use `checkoutStep` in this case. I feel it is present for that exact reason. If not, what is the point of it? If every other step before payment information sets the step properly we should not have any corrupted step state.

Reply:
Also payment information is plugable step. Credit card has diff data and Paypal in the future will have diff data. What should the `usePaymentInformation.js` talon look for to determine if data is actually present. We can not write if-else cases to look for diff data for diff payment methods. Either have local variable called `doneEditing` and set it to true on`handlePaymentSuccess` or use `checkoutStep`.